subprojects/docs/src/docs/userguide/migrating_from_maven.adoc

```diff
@@ -0,0 +1,518 @@
+To add profile-like behavior, first create build files for each environment in the project root: `profile-default.gradle`, `profile-test.gradle`, and `profile-prod.gradle`.
+Next, add a condition similar to the following to the main build file:
+
+[source,groovy]
+----
+if (!hasProperty('buildProfile')) ext.buildProfile = 'default'
+apply from: "profile-${buildProfile}.gradle"
+----
+
+All you have to do then is put the environment-specific configuration, such as project properties, dependencies, etc., in the corresponding build file.
+To activate a particular profile, you can just pass in the relevant project property on the command line:
+
+[listing.terminal]
+----
+$ gradle -PbuildProfile=test build
+----
+
+Or you can set the project property another way.
+It's up to you.
+And those conditions don't just have to check project properties.
+You could check environment variables, the JDK version, the OS the build is running on, and anything else you can imagine.
+
+One thing to bear in mind is that high level ‘if' statements make builds harder to understand and maintain, similar to the way they complicate Object-Oriented code.
+The same applies to profiles.
+Gradle offers you many better ways to avoid the extensive use of profiles that Maven often requires, for example by offering variants.
+
+For a lengthier discussion on working with Maven profiles in Gradle, look no further than http://gradle.org/feature-spotlight-gradles-support-maven-pom-profiles[this article] by Benjamin Muschko.
+
+
+== Filtering resources
+
+Maven has a phase called `process-resources` that has the goal `resources:resources` bound to it by default.
+This gives the build author an opportunity to perform variable substitution on various files, such as web resources, packaged properties files, etc.
+
+The Java plugin for Gradle provides a `processResources` task to do the same thing.
+Here's an example configuration:
+
+.Filtering resources via the `processResources` task
+====
+include::sample[dir="userguide/mavenMigration/resourceFiltering/groovy",files="build.gradle[tags=process-resources]"]
+====
+
+So the left hand side of each colon is the token name and the right hand side is a project property.
+This variable substitution will apply to all your resource files (the ones under `src/main/resources` usually).
+
+Gradle has other powerful ways for property processing.
+You can hook in your own filter via a closure that allows you to process the content line by line, or you can add your own `FilterReader` implementation.
+For more details, see the documentation for the link:{javadocPath}/org/gradle/api/file/ContentFilterable.html[ContentFilterable] interface which all copy and archive tasks implement.
+
+
+== Configuring integration tests
+
+Although unit tests are very useful, they can't ensure that an application or library works as a whole.
```

Comment:
:o: I think we can omit this entire paragraph.

Reply:
Probably right. Done.